test/test-helpers/did-render.ts

```diff
@@ -0,0 +1,13 @@
+import Application from '../../src/application';
+
+async function didRender(app: Application): Promise<void> {
+  return new Promise<void>(resolve => {
+    let watcher = setInterval(function() {
+      if (app['_scheduled']) return;
```

Comment:
I'm slightly concerned that when _scheduled is set back to false, the render is not actually completed. I know that right now we have sync render and sync rerender, but I believe that we all generally have plans to "yield to the browser" if things are taking too long (ala react fiberish). At that point checking _scheduled is probably not enough.